subprojects/wrapper/src/main/java/org/gradle/wrapper/Download.java

```diff
@@ -97,6 +102,8 @@ private void downloadInternal(URI address, File destination)
 
                 out.write(buffer, 0, numRead);
             }
+        } catch (SocketTimeoutException e) {
+            throw new IOException("Downloading from " + address + " failed: timeout", e);
```

Comment:
I changed this to use `safeUrl` instead of `address`, since `address` can contain user credentials and we don't want to log those.